File: frontend/src/services/monitor.ts
import { api } from './api';
import type { MonitorStatus } from '../types/settings';

export const monitorApi = {
  getStatus: async (): Promise<MonitorStatus> => {
    const response = await api.get('/monitor/status');
    return response.data;
  },

  startMonitoring: async (): Promise<{ message: string; status: MonitorStatus }> => {
    const response = await api.post('/monitor/start');
    return response.data;
  },

  stopMonitoring: async (): Promise<{ message: string; status: MonitorStatus }> => {
    const response = await api.post('/monitor/stop');
    return response.data;
  },
};

File: backend/api/routes/monitor.py
from fastapi import APIRouter, HTTPException, Depends

router = APIRouter()

# Import the global email monitor instance
def get_email_monitor():
    from main import email_monitor
    return email_monitor

@router.get("/status")
async def get_monitoring_status(monitor = Depends(get_email_monitor)):
    """Get current monitoring status"""
    return {"is_monitoring": monitor.is_running}

@router.post("/start")
async def start_monitoring(monitor = Depends(get_email_monitor)):
    """Start email monitoring"""
    try:
        await monitor.start_monitoring()
        return {"message": "Monitoring started", "status": {"is_monitoring": monitor.is_running}}
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"Error starting monitoring: {str(e)}")

@router.post("/stop")
async def stop_monitoring(monitor = Depends(get_email_monitor)):
    """Stop email monitoring"""
    try:
        await monitor.stop_monitoring()
        return {"message": "Monitoring stopped", "status": {"is_monitoring": monitor.is_running}}
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"Error stopping monitoring: {str(e)}")